NodeManager.cpp

```diff
@@ -1040,6 +1040,86 @@ void SensorDHT::onReceive(const MyMessage & message) {
 #endif
 
 /*
+   SensorAM2320
+*/
+#if MODULE_AM2320 == 1
+// contructor
+SensorAM2320::SensorAM2320(int child_id, int sensor_type): Sensor(child_id,A2) {
+
+  // store the sensor type (0: temperature, 1: humidity)
+  _sensor_type = sensor_type;
+  if (_sensor_type == SensorAM2320::TEMPERATURE) {
+    // temperature sensor
+    setPresentation(S_TEMP);
+    setType(V_TEMP);
+    setValueType(TYPE_FLOAT);
+  }
+  else if (_sensor_type == SensorAM2320::HUMIDITY) {
+    // humidity sensor
+    setPresentation(S_HUM);
+    setType(V_HUM);
+    setValueType(TYPE_FLOAT);
+  }
+
+}
+
+// what do to during before
+void SensorAM2320::onBefore() {
+
+}
+
+// what do to during setup
+void SensorAM2320::onSetup() {
+}
+
+// what do to during loop
+void SensorAM2320::onLoop() {
+    _th = new AM2320();
```

Comment:
Shall we move this into the constructor? Otherwise a new object is created at every loop()